include/aws/common/system_resource_util.h

```diff
@@ -0,0 +1,30 @@
+#ifndef AWS_COMMON_SYSTEM_RESOURCE_UTIL_H
```

Comment:
Debatable: We already have a system_info header which might be a good place for these functions.

Reply:
Im on the fence about it. I was originally planning to put it into system_info, but functions here have a slightly different use case - how many resource is my process consuming vs which system im running on. I was also planning to add functions to modify resource limits to this file, which will diverge things even more in future. So im leaning towards a separate header.